### ISSUE TITLE
Disable option in SelectInput

### ISSUE THREAD

**Issue by svrcekmichal** [marmelab/react-admin#2512]
**Is your feature request related to a problem? Please describe.**
I've added list of user statuses into our admin page, and transition to few of them should be not possible, the problem is that it's not possible to mark option as disabled. This is supported in material ui, since `MenuItem` has prop `disabled` but `SelectInput` is not passing that prop down

**Describe the solution you'd like**
```
<SelectInput source="status" choices={[
  { id: "DISABLED", name: "disabled", disabled: true},
  { id: "ENABLED", name:"enabled"}
]}
```

**Comment by fzaninotto** [comment]
Nice enhancement suggestion, and quite easy to add. Would you like to give it a try?

**Comment by mikaoelitiana** [comment]
I can try this if it's still available.

**Comment by djhi** [comment]
It is, please do :)

**Comment by mikaoelitiana** [comment]
I created a PR, I hope it's clear.

**Comment by fzaninotto** [comment]
Fixed by #2555
